ledger-core/v2/virtual/object/object.go

```diff
@@ -255,6 +263,11 @@ func (sm *SMObject) createWaitPendingOrderedSM(ctx smachine.ExecutionContext) {
 }
 
 func (sm *SMObject) stepSendVStateReport(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	if sm.migrationCount > 0 {
+		sm.migrationCount++
```

Comment:
1. misleading name
2. sm.migrationCount++ is called inside and outside of if - why not to call it before if?

Reply:
Fix name and counter increment